src/Mobs/Path.cpp

```diff
@@ -251,72 +212,87 @@ bool cPath::StepOnce()
 	// Check North, South, East, West at our own height or below. We are willing to jump up to 3 blocks down.
 
 
-	if (!done_east)
+	if (!DoneEast)
 	{
-		for (int i = 0; i >= -3; --i)
+		for (int y = 0; y >= -3; --y)
 		{
-			if (ProcessIfWalkable(CurrentCell->m_Location + Vector3i(1, i, 0),  CurrentCell, 10))
+			if (ProcessIfWalkable(CurrentCell->m_Location + Vector3i(1, y, 0),  CurrentCell, NORMAL_G_COST))
 			{
-				done_east = true;
+				DoneEast = true;
+				if (y == 0)
+				{
+					WalkableEast = true;
+				}
 				break;
 			}
 		}
 	}
 
-	if (!done_west)
+	if (!DoneWest)
 	{
-		for (int i = 0; i >= -3; --i)
+		for (int y = 0; y >= -3; --y)
 		{
-			if (ProcessIfWalkable(CurrentCell->m_Location + Vector3i(-1, i, 0),  CurrentCell, 10))
+			if (ProcessIfWalkable(CurrentCell->m_Location + Vector3i(-1, y, 0),  CurrentCell, NORMAL_G_COST))
 			{
-				done_west = true;
+				DoneWest = true;
+				if (y == 0)
+				{
+					WalkableWest = true;
+				}
 				break;
 			}
 		}
 	}
 
-	if (!done_south)
+	if (!DoneSouth)
 	{
-		for (int i = 0; i >= -3; --i)
+		for (int y = 0; y >= -3; --y)
 		{
-			if (ProcessIfWalkable(CurrentCell->m_Location + Vector3i(0, i, 1),  CurrentCell, 10))
+			if (ProcessIfWalkable(CurrentCell->m_Location + Vector3i(0, y, 1),  CurrentCell, NORMAL_G_COST))
 			{
-				done_west = true;
+				DoneWest = true;
```

Comment:
This should definitely be south. I am shocked Coverity discovered this. Since the DoneX variables are currently not used these blocks of code, the bug is dormant and had no ingame effect.